### ISSUE TITLE
Ensure all files in R/ end in .R (not .r)

### ISSUE THREAD

**Issue by hadley** [tidyverse/ggplot2#3640]

**Comment by carywreams** [comment]
Questions:

+ straight-forward source filename changes ?
+ also change relevant ```@include``` references within these files ?
+ any other anticipated sympathetic changes (such as ```@include```) in other directories?
+ what about similar changes to files in ```tests/testthat/``` ?
+ what else should I look for ?

**Comment by carywreams** [comment]
Gonna wrap this and create PR without changing any filenames in ```testthat/```.

testthat results precisely match those on master.
+ [on_i3640.txt](https://github.com/tidyverse/ggplot2/files/3928273/on_i3640.txt)
+ [on_master.txt](https://github.com/tidyverse/ggplot2/files/3928274/on_master.txt)

**Comment by carywreams** [comment]
My notes:
```
# just change the .r files
for f in *.r;
do
    mv "$f" "${f%.r}.R"
done

# in-place changes to @include statements
# with a .r filename
# changes .r to .r.R then changes .r.R to .R
sed -i 's/@include.*\.r/&.R/g;s/\.r\.R/.R/g' *R
```

**Comment by yutannihilation** [comment]
While I guess no one is against this suggestion, in practice, this is very difficult because the pull requests easily get conflicted. Maybe we need some strategy to get this done but I have no idea...

**Comment by clauswilke** [comment]
Yes, in particular when there are PRs pending that touch a large number of files, e.g. #2749. We'd have to first triage all PRs and close or merge as many as possible by a certain date, then not create any new ones past that date or only PRs that touch files ending in .R, then rename, then open the branch again for submission. This is possible but not trivial.

**Comment by carywreams** [comment]
...and not sure this type of change warrants "good will capital."

**Comment by olivroy** [comment]
While looking at old issues, I saw that this was completed in #5221

**Comment by teunbrand** [comment]
Thanks for the reminder @olivroy!
The renaming personally caused a few headaches for some pending PRs, but I can live with it.
